doc/developer/platform/formalism.md

```diff
@@ -0,0 +1,176 @@
+    The partial TVC can now be used in queries, though results may not be immediately available.
+
+2. Repeatedly, either of the these two steps occur:
+
+    a. The `upper` frontier advances, revealing more of the future of the pTVC.
+
+    b. The `since` frontier advances, concealing some of the past of the pTVC.
+
+3. Eventually, the `since` frontier advances to the empty collection, rendering the TVC complete and closed.
+
+The `GlobalId` is never reused.
+
+# Storage and Compute
+
+Materialize produces and maintains time-varying collections in two distinct ways:
+
+*   The **Storage** layer records explicit TVCs in response to direct instruction (e.g. user DML), integrations with external sources of data, and feedback from Materialize itself.
+    The representation in the Storage layer is as explicit update triples and maintained `since` and `upper` frontiers.
+    The primary property of the Storage layer is that, even across failures, it is able to reproduce all updates for times that lie between `since` and `upper` frontiers that it has advertised.
+
+*   The  **Compute** layer uses a view definition to transform input TVCs into the exactly corresponding output TVCs.
+    The output TVC varies *exactly* with the input TVCs:
+    For each `time`, the output collection at `time` equals the view applied to the input collections at `time`.
+
+The Storage and Compute layers both track `since` and `upper` frontiers for the collections they manage.
+For both layers, the `since` frontier is controlled by their users and it does not advance unless it is allowed.
+The `upper` frontier in Storage is advanced when it has sealed a frontier and made the data up to that point durable.
+The `upper` frontier in Compute is advanced when its inputs have advanced and its computation has pushed updates fully through the view definitions.
+
+## Capabilities
+
+The `since` and `upper` frontiers are allowed to advance through the use of "capabilities".
+
+* `ReadCapability(id, frontier)` prevents the `since` frontier associated with `id` from advancing beyond `frontier`.
+* `WriteCapability(id, frontier)` prevents the `upper` frontier associated with `id` from advancing beyond `frontier`.
+
+A capability presents the `id` and `frontier` values, but is identified by a system-assigned unique identifier.
+Any user presenting this identifier has the associated capability to read from or write to the associated pTVC.
+Actions which read or write will validate the capability (e.g. in case a lease has expired) and may return errors if the capability is not valid as stated.
+
+Capabilities are durable, and their state is tracked by the system who associates a unique identifier with each.
+Capabilities can be cloned and transfered to others, providing the same rights (and responsibilities).
+Capabilities can be downgraded to frontiers that are greater or equal to the held frontier, which may then allow the `since` or `upper` frontier to advance.
```

Comment:
What will be a good use-case for a user downgrading the capability to a frontier, as opposed to just dropping the capability? Is this needed just before exercising the corresponding read/write ability because the `frontier` element cannot be accessed otherwise? 

If this is correct, it means that each capability models the ability for a single "read" or "write" request. If this is the case the 1:1 correspondence is not obvious from the current text.

Reply:
The coordinator will want to hold a read capability for each of the collections it wants to read, as otherwise it cannot be sure that it will have read access (we might just close out the collection). However, it wants to regularly downgrade that capability to allow compaction. Similarly with write capabilities, the coordinator wants to write at various times, and slowly advance the capability to reveal that a subset of times are now bound.